text/2020-01-30-key-spaces.md

```diff
@@ -0,0 +1,181 @@
+# Title
+
+Key Spaces
+
+## Summary
+
+When multiple applications store data in TiKV, we need a clear pattern for avoiding conflicting use of the key space in TiKV.
+TiKV is a monolithic KV store, so each application must choose a key prefix.
+Key Spaces formalizes this process by registering a key space for an application in TiKV itself.
+
+When combined with authentication (not part of this proposal), this can secure an application's key space.
+
+Additionally, key spaces make it possible for TiKV to avoid conflicts during a data restore by changing the key prefix assigned to a key space.
```

Comment:
Can you give a detailed example for restore?

Reply:
Did you see the section "Backup and Restore" ?

Reply:
Can you explain why need to change the prefix?

Reply:
For example if someone wanted to combine 2 separate TiDB clusters into one TiKV cluster. They could have a collision which can be ensuring that when restored they use different key prefix

Reply:
> combine 2 separate TiDB clusters into one TiKV cluster

This sounds interesting, I don't know if this demand exists, @shenli what's your opinion for it.

Reply:
I explained this a little better in the RFC now.